cmd/juju/cloud/addcredential_test.go

```diff
@@ -142,14 +149,166 @@ func (s *addCredentialSuite) TestAddFromFileNoCredentialsFound(c *gc.C) {
 }
 
 func (s *addCredentialSuite) TestAddFromFileExisting(c *gc.C) {
+	s.authTypes = []jujucloud.AuthType{jujucloud.UserPassAuthType, jujucloud.AccessKeyAuthType}
 	s.store.Credentials = map[string]jujucloud.CloudCredential{
 		"somecloud": {
 			AuthCredentials: map[string]jujucloud.Credential{"cred": {}},
 		},
 	}
 	sourceFile := s.createTestCredentialData(c)
 	_, err := s.run(c, nil, "somecloud", "-f", sourceFile)
-	c.Assert(err, gc.ErrorMatches, `local credentials for cloud "somecloud" already exist; use --replace to overwrite / merge`)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(s.store.Credentials, jc.DeepEquals, map[string]jujucloud.CloudCredential{
+		"somecloud": {
+			AuthCredentials: map[string]jujucloud.Credential{
+				"cred": {},
+				"me": jujucloud.NewCredential(jujucloud.AccessKeyAuthType, map[string]string{
+					"access-key": "<key>",
+					"secret-key": "<secret>",
+				})},
+		},
+	})
+}
+
+func (s *addCredentialSuite) TestAddInvalidRegionSpecified(c *gc.C) {
+	s.authTypes = []jujucloud.AuthType{jujucloud.AccessKeyAuthType}
+	_, err := s.run(c, nil, "somecloud", "--region", "someregion")
+	c.Assert(err, gc.ErrorMatches, `provided region "someregion" for cloud "somecloud" not valid`)
+}
+
+func (s *addCredentialSuite) testCloudWithRegions(c *gc.C) {
```

Comment:
This seems like a poor name. This isn't a test, but named *like* a test (although unexported). setupCloudWithRegions or initCloudWithRegions maybe.

Reply:
Prefixed with 'setup' instead.